src/components/NavItem/NavItem.tsx

```diff
@@ -0,0 +1,129 @@
+/* External dependencies */
+import React, { useCallback, useMemo, memo } from 'react'
+import { noop, isNil } from 'lodash-es'
+
+/* Internal dependencies */
+import useDepthContext, { DepthContext } from 'Hooks/useDepthContext'
+import { IconSize, isIconName } from 'Components/Icon'
+import type NavItemProps from './NavItem.types'
+import {
+  ItemWrapper,
+  LeftIconWrapper,
+  StyledIcon,
+  ChevronWrapper,
+  ContentWrapper,
+  RightContentWrapper,
+} from './NavItem.styled'
+
+const DEFAULT_INDENT = 30
+const DEFAULT_PADDING_LEFT = 6
+export const NAV_ITEM_TEST_ID = 'bezier-react-nav-item'
+
+function NavItem(
+  {
+    as,
+    testId = NAV_ITEM_TEST_ID,
+    name,
+    style,
+    className,
+    interpolation,
+    children,
+    content,
+    rightContent,
+    leftIcon,
+    leftIconColor = 'txt-black-dark',
+    indent = DEFAULT_INDENT,
+    active,
+    open,
+    onClick = noop,
+  }: NavItemProps,
+) {
+  const depthInfo = useDepthContext((parent) => ({
+    depth: parent.depth + 1,
+    paddingLeft: (parent.paddingLeft ?? (DEFAULT_PADDING_LEFT - indent)) as number + indent,
+  }))
+
+  const handleClickItem = useCallback((e?: React.MouseEvent) => {
+    onClick(e, name)
+  }, [
+    name,
+    onClick,
+  ])
+
+  const leftComponent = useMemo(() => {
+    if (!isNil(leftIcon) && isIconName(leftIcon)) {
```

Comment:
```suggestion
    if (isIconName(leftIcon)) {
```
이렇게만 guard 해도 될 것 같아요

+) isIconName이 Null check도 해야 할 것 같아요

Reply:
```
export const isIconName = (name): name is IconName => Object.keys(icons).includes(name)
```
여기서 알아서 null도 걸러질 것 같네요